src/component/tag-types/form-tag-type-component.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Textfield, Card, CardTitle, CardText, CardActions } from 'react-mdl';
+import { FormButtons, styles as commonStyles } from '../common';
+import { trim } from '../common/util';
+class AddTagTypeComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            tagType: props.tagType,
+            errors: {},
+            dirty: false,
+            currentLegalValue: '',
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (!state.tagType && props.tagType.name) {
+            return { tagType: props.tagType };
+        } else {
+            return null;
+        }
+    }
+
+    validateName = async name => {
+        const { errors } = this.state;
+        const { validateName } = this.props;
+        try {
+            await validateName(name);
+            errors.name = undefined;
+        } catch (err) {
+            errors.name = err.message;
+        }
+        this.setState({ errors });
+    };
+
+    setValue = (field, value) => {
+        const { tagType } = this.state;
+        tagType[field] = value;
+        this.setState({ tagType, dirty: true });
+    };
+
+    onCancel = evt => {
+        evt.preventDefault();
+        this.props.history.push('/tag-types');
+    };
+
+    onSubmit = async evt => {
+        evt.preventDefault();
+        const { tagType } = this.state;
+        await this.props.submit(tagType);
+        this.props.history.push('/tag-types');
```

Comment:
should only do this if submit does not fail.

Reply:
True